src/script/script.cpp

```diff
@@ -15,144 +15,157 @@ const char* GetOpName(opcodetype opcode)
+	case OP_MUL                    : return "OP_MUL";
+	case OP_DIV                    : return "OP_DIV";
+	case OP_MOD                    : return "OP_MOD";
+	case OP_LSHIFT                 : return "OP_LSHIFT";
+	case OP_RSHIFT                 : return "OP_RSHIFT";
+	case OP_BOOLAND                : return "OP_BOOLAND";
+	case OP_BOOLOR                 : return "OP_BOOLOR";
+	case OP_NUMEQUAL               : return "OP_NUMEQUAL";
+	case OP_NUMEQUALVERIFY         : return "OP_NUMEQUALVERIFY";
+	case OP_NUMNOTEQUAL            : return "OP_NUMNOTEQUAL";
+	case OP_LESSTHAN               : return "OP_LESSTHAN";
+	case OP_GREATERTHAN            : return "OP_GREATERTHAN";
+	case OP_LESSTHANOREQUAL        : return "OP_LESSTHANOREQUAL";
+	case OP_GREATERTHANOREQUAL     : return "OP_GREATERTHANOREQUAL";
+	case OP_MIN                    : return "OP_MIN";
+	case OP_MAX                    : return "OP_MAX";
+	case OP_WITHIN                 : return "OP_WITHIN";
+
+	// crypto
+	case OP_RIPEMD160              : return "OP_RIPEMD160";
+	case OP_SHA1                   : return "OP_SHA1";
+	case OP_SHA256                 : return "OP_SHA256";
+	case OP_HASH160                : return "OP_HASH160";
+	case OP_HASH256                : return "OP_HASH256";
+	case OP_CODESEPARATOR          : return "OP_CODESEPARATOR";
+	case OP_CHECKSIG               : return "OP_CHECKSIG";
+	case OP_CHECKSIGVERIFY         : return "OP_CHECKSIGVERIFY";
+	case OP_CHECKMULTISIG          : return "OP_CHECKMULTISIG";
+	case OP_CHECKMULTISIGVERIFY    : return "OP_CHECKMULTISIGVERIFY";
+
+	// expansion
+	case OP_NOP1                   : return "OP_NOP1";                  // OP_NOP1
+	case OP_CHECKLOCKTIMEVERIFY    : return "OP_CHECKLOCKTIMEVERIFY";   // OP_NOP2
+	case OP_NOP3                   : return "OP_NOP3";                  // OP_NOP3
+	case OP_NOP4                   : return "OP_NOP4";                  // OP_NOP4
+	case OP_NOP5                   : return "OP_NOP5";                  // OP_NOP5
+	case OP_NOP6                   : return "OP_NOP6";                  // OP_NOP6
+	case OP_NOP7                   : return "OP_NOP7";                  // OP_NOP7
+	case OP_NOP8                   : return "OP_NOP8";                  // OP_NOP8
+	case OP_NOP9                   : return "OP_NOP9";                  // OP_NOP9
+	case OP_NOP10                  : return "OP_NOP10";                 // OP_NOP10
+
+	// zerocoin
+	case OP_ZEROCOINMINT           : return "OP_ZEROCOINMINT";
+	case OP_ZEROCOINSPEND          : return "OP_ZEROCOINSPEND";
+	case OP_ZEROCOINPUBLICSPEND    : return "OP_ZEROCOINPUBLICSPEND";
+
+	// memory store
+	case OP_MLOAD				   : return "OP_MLOAD";
+	case OP_MSTORE				   : return "OP_MSTORE";
+	case OP_MSTORE8				   : return "OP_MSTORE8";
+
+	// storage
+	case OP_SLOAD				   : return "OP_SLOAD";
+	case OP_SSTORE				   : return "OP_SSTORE";
+
+	// smart contract actions
+	case OP_SC_PUBLISH			   : return "OP_SC_PUBLISH";
+	case OP_SC_RUN			   	   : return "OP_SC_RUN";
```

Comment:
Why not only OP_PUBLISH and OP_RUN? SC seems redundant

Reply:
At first, I do it like OP_PUBLISH and OP_RUN. Then I thought that in the future we may need OP_RUN for another coding so decided to branch them out as OP_SC_PUBLISH and OP_SC_RUN.
OP_PUBLISH may be used in Proof of Ownership as well (to publish data into a database) in the future for instance.

But it's ok. Let's make it as OP_PUBLISH and OP_RUN and if we will need other PUBLISH or RUN, we can branch out them in the future like OP_OWNERSHIP_PUBLISH etc.